wallet/wallet.go

```diff
@@ -288,19 +330,42 @@ func (w *Wallet) AgendaChoices(ctx context.Context) (choices []AgendaChoice, vot
 	if err != nil {
 		return nil, 0, errors.E(op, err)
 	}
+	if ticketHash != nil && !ownTicket {
+		return nil, 0, errors.E("ticket not owned by wallet")
```

Comment:
this should have a errors.NoExist code if anything, though you should get that anyways once you use an updated udb function to query the preferences.